src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -251,21 +256,52 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name,
             if config_present is False:
                 raise CLIError("Helm release named 'azure-arc' is already present but the azure-arc agent pods are either missing or deployed unsuccessfully.")
     
-    # Adding helm repo
-    if kube_context is None:
-        cmd1 = ["helm", "repo", "add", "azurearcfork8s", "https://azurearcfork8s.azurecr.io/helm/v1/repo", "--kubeconfig", kube_config]
+    # Retrieving Helm chart OCI Artifact location
+    repository_path = None
+    cred, _, _ = profile.get_login_credentials(
+        resource='https://management.core.windows.net/')
+    token = cred._token_retriever()[2].get('accessToken')
+
+    get_chart_location_url = "https://{}.dp.kubernetesconfiguration.azure.com/{}/GetLatestHelmPackagePath?api-version=2019-11-01-preview".format(location, 'azure-arc-k8sagents')
+    query_parameters = {}
+    query_parameters['releaseTrain'] = 'stable'
+    header_parameters = {}
+    header_parameters['Authorization'] = "Bearer {}".format(str(token))
+    try:
+        response = requests.post(get_chart_location_url, params=query_parameters, headers=header_parameters)
+    except Exception as e:
+        raise CLIError("Error while fetching helm chart repository path: " + str(e))
+    if response.status_code == 200:
```

Comment:
nitpick: Do you want to retry the config dp call in case of 429 (throttling scenario)

Reply:
Not expecting throttling for private preview. Hence will not add for this PR.